monix-eval/shared/src/main/scala/monix/eval/internal/TaskFromFuture.scala

```diff
@@ -42,14 +46,14 @@ private[eval] object TaskFromFuture {
             rawAsync(startSimple(_, _, f))
         }
       case Some(value) =>
-        Task.fromTry(value)
+        Task.async(cb => cb(value))
```

Comment:
It sucks but I don't know if there is any way to detect whether `Future` has been built using `successful` or not

Reply:
Pretty sure those futures have `.value` set to `Some(try)`

Reply:
`Future` crossing async boundaries can also be  `Some(try)` if it finished earlier